### PR TITLE
feat(engine): implement ADR-0036 D-PLACEMENT bundle-placement engine (H4)

### PR DESCRIPTION
**Pull request by Szczepanov**
## Summary

Second of the 3-PR split for ADR-0036's D-PLACEMENT decision (H4) -- D-WINDOW's model
landed in [#429](https://github.com/Szczepanov/adaptive-training-recommender/pull/429).
This PR delivers **D-PLACEMENT's bundle-placement engine only**, as a pure, unwired
module, per the same pattern D-TIME/D-LEDGER/D-WINDOW each shipped.

## What this PR delivers

[`app/src/engine/intradayBundlePlacement.ts`](app/src/engine/intradayBundlePlacement.ts)
resolves one v4 intraday bundle's requested windows against real `ScheduleWindow`
availability, checks the combined minute/systemic-cost budget via `dailyLedger.ts`,
respects ADR-0035 rest and fixed commitments, and computes scheduled separation from
D-TIME's resolved instants -- an atomic confirmed-proposal API analogous to
`externalPlacement.ts`'s `proposeReplacement`/`applyConfirmedProposal`, but for whole
same-date bundles:

- **`proposeBundlePlacement`** never returns a partial placement: either every member
  gets a resolved window binding, or the whole proposal is `infeasible` with a reason.
  Admission is evaluated sequentially in `order`, so an earlier member's consumption
  correctly reduces what a later member can draw from the same day's shared ledger. A
  date with no persisted `ScheduleWindow`s falls back to one synthetic whole-day slot
  (D-WINDOW's legacy case); since at most one occurrence binds to any one slot, a bundle
  needing two-or-more windows on such a date is correctly infeasible rather than
  fabricating an AM/PM pair from missing metadata.
- **`dropOptionalBundleMember`** is the athlete's explicit, non-automatic drop action —
  it throws for a required or already-started member, and D-SCHEMA's existing "no
  required session depends on an optional predecessor" invariant means it can never
  orphan a required one.
- **`confirmBundlePlacement`** is the sole confirmation boundary, mirroring
  `applyConfirmedProposal`; it throws on an infeasible proposal rather than confirming a
  partial one, and writes nothing — no persisted bundle-binding store exists yet (that's
  wiring/D-AUDIT's job, not this PR).

A member that has already started always carries its existing binding through unchanged
and keeps its window consumed for the rest of the bundle ("once a member starts, do not
move its history").

## Investigated: interaction with PR #428 (merged the same day as D-WINDOW)

`ScheduleOverlay` now feeds `resolveAvailability`'s whole-day minute/cost ceiling. No
conflict: this engine's `ledger` input is already an opaque `DailyLedgerResult` supplied
by the caller, so whatever narrows the day's ceiling (fixed activities, `ScheduleOverlay`,
or both) is transparent to this module.

## Tests

`intradayBundlePlacement.test.ts` (22 cases) covers the ADR's D-PLACEMENT-relevant
deterministic cases: AM/PM placement into distinct windows, the legacy-date single-slot
restriction, authored rest closing every window, fixed-activity occupancy (both a
known-clock-time activity blocking only its overlapping window and an unknown-clock-time
one blocking the whole date), the 90-minute-ceiling/60-minute-AM worked example extended
to a real bundle, exhausted-systemic-cost rejection with spare minutes still available,
separation success/failure, started-member preservation, and a nonexistent Warsaw
spring-forward boundary correctly reported as unresolved.

**One test caught a real bug before merge**: `elapsedMinutesBetweenInstants(startInstant,
endInstant)` computes `start - end` (per its own doc comment), not `end - start` — my
first draft passed the predecessor's end and the dependent's start in the wrong order,
which happened to still pass the "insufficient separation" test (a negative gap is always
less than any positive requirement) but correctly failed the "sufficient separation"
test. Fixed.

## Not in scope here

- Not wired into `evaluateTrainingWithIntent` or any decision path.
- `POLICY_VERSION` unchanged; `check-policy-drift.mjs origin/main` confirms zero
  decision-affecting files touched; `simulate:diff` shows the same pre-existing,
  unrelated baseline staleness already flagged separately (tracked as its own background
  task).
- No persisted bundle-binding Firestore store (D-AUDIT / wiring PR's job).
- D-REASSESS's actual-performed-timestamp recomputation is explicitly not implemented
  here — this module's "separation" check uses *scheduled* window boundaries, since no
  session has started yet at placement time.

## Next PR (3 of 3)

Wire D-PLACEMENT's engine (and the ledger's remainder/admission math generally) into
`evaluateTrainingWithIntent`, with a real `POLICY_VERSION` bump and full
scenario/simulate-diff verification of the actual behavior change. Then D-REASSESS/
D-AUDIT.

## Verification

```
npm run typecheck   # pass
npm run lint         # pass
npx vitest run src/engine/intradayBundlePlacement.test.ts  # 22 passed
npx vitest run       # 3673 passed, 190 skipped (full suite)
npm run build        # pass
npm run test:rules   # 190 passed (Firestore emulator; no rules changes in this PR)
npm run simulate:scenarios && npm run simulate:diff   # only pre-existing, unrelated baseline drift
node scripts/check-policy-drift.mjs origin/main   # PASSED: no decision-affecting files modified
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added scheduling support for evaluating intraday bundle activities across available time windows.
  - Accounts for rest days, existing activities, dependencies, time constraints, capacity limits, and started activities.
  - Supports optional activity removal when placement is infeasible and validates proposals before confirmation.

- **Documentation**
  - Updated planning and handoff documentation to describe placement behavior and remaining runtime integration work.

- **Tests**
  - Added comprehensive coverage for placement rules, validation failures, conflicts, backtracking, and edge cases.

### CODE REVIEW

File: app/src/engine/intradayBundlePlacement.ts [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+/** The date's fixed-activity-occupied intervals. An activity with a known `startTime`
+ * only blocks windows it actually overlaps; one without a resolvable clock time is
+ * conservative and blocks the whole date, matching legacy single-slot placement's
+ * existing whole-date fixed-activity precedent (`externalPlacement.ts`'s
+ * `PlacementOccupancy`) when finer-grained information is unavailable. */
+function fixedOccupiedIntervals(
+    date: string,
+    fixedActivities: readonly FixedActivity[],
+): readonly { startLocal: string; endLocal: string }[] | 'whole-date' {
+    const activities = fixedActivities.filter(activity => activity.date === date && !activity.isCompleted);
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayBundlePlacement.ts]: Confirmed and fixed in the latest commit. `fixedOccupiedIntervals` was missing the `activity.fixed` check, so a movable placeholder blocked placement the same as an immovable commitment, even though it's exactly the kind of activity the athlete could reschedule around. Added `activity.fixed &&` to the filter and a regression test confirming a non-fixed activity no longer blocks placement.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: app/src/engine/intradayBundlePlacement.ts [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    const occupied = fixedOccupiedIntervals(date, fixedActivities);
+    let pool = resolvePlacementWindows(date, scheduleWindows).filter(window => !windowBlockedByFixed(window, occupied));
+
+    const bindings = new Map<string, ResolvedWindowBinding>();
+    let runningLedger = ledger;
+
+    for (const member of sorted) {
+        if (member.started && member.existingBinding) {
+            bindings.set(member.sessionId, member.existingBinding);
+            pool = pool.filter(window => window.windowId !== member.existingBinding!.windowId);
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayBundlePlacement.ts]: Confirmed and fixed in the latest commit. Started-window exclusion now filters the pool by interval overlap against the existing binding's `boundStartLocal`/`boundEndLocal` (reusing `scheduleWindowsOverlap`), not the binding's stored `windowId`. Added a regression test where the current `ScheduleWindow` set uses a different id for the same 06:00-07:00 interval the member actually started in, confirming that interval still correctly stays excluded from the pool.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: app/src/engine/intradayBundlePlacement.ts [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+            continue;
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayBundlePlacement.ts]: Confirmed and fixed in the latest commit. A started member's `estimatedMinutes`/`estimatedSystemicCost` are now debited from the running ledger via the same `debitLedger` helper unstarted members use, before evaluating later members -- closing exactly the double-admission gap you described. I also tightened `proposeBundlePlacement`'s doc comment: `ledger` must now be the day's remainder *before any* of this bundle's own members (started or not), since this function debits all of them itself; a caller that had already subtracted a started member's consumption would now double-charge it, so that contract needed to be explicit either way. Added a regression test with a 90-minute ceiling, a started 60-minute member, and a new 40-minute member, confirming only 30 minutes remain and the bundle is correctly infeasible.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: app/src/engine/intradayBundlePlacement.ts [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+        if (candidates.length === 0) {
+            return infeasible(
+                bundleId,
+                `Session '${member.sessionId}' requests ${member.requestedWindow.startLocal}-${member.requestedWindow.endLocal} on ${date}, `
+                + 'but no available window fits its duration after fixed commitments and other bundle members.',
+            );
+        }
+
+        const chosen = candidates[0];
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayBundlePlacement.ts]: Confirmed and fixed in the latest commit. Replaced the greedy `candidates[0]` selection with a recursive backtracking search (`assignFrom`): each unstarted member tries its candidates in best-overlap order, but if a choice leaves no valid assignment for the remaining members, it backtracks and tries that member's next candidate before the whole bundle is reported infeasible -- including re-checking separation and ledger admission per candidate, since those depend on which window ends up bound. Added a regression test with exactly your scenario shape (an earlier member fits two windows, a later member fits only one of them) confirming the engine now finds the complete assignment instead of rejecting the bundle from the first greedy choice.

I kept this as plain recursion rather than full bipartite matching since bundle/window counts are small in practice (first release scope, per the ADR) and the state per branch (ledger remainder, separation against already-bound predecessors) is naturally handled by DFS with backtracking. Noted the worst-case exponential branching in the function's doc comment.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
